gpflow/posteriors.py

```diff
@@ -123,13 +120,89 @@ def __init__(
         instantiate subclasses, developers need to ensure their subclasses
         don't change the constructor signature.
         """
-        self.inducing_variable = inducing_variable
+        super().__init__()
+
         self.kernel = kernel
+        self.X_data = X_data
+        self.alpha = alpha
+        self.Qinv = Qinv
+        self.mean_function = mean_function
+
+    def _add_mean_function(self, Xnew, mean):
+        if self.mean_function is None:
+            return mean
+        else:
+            return mean + self.mean_function(Xnew)
+
+    def fused_predict_f(
+        self, Xnew, full_cov: bool = False, full_output_cov: bool = False
+    ) -> MeanAndVariance:
+        """
+        Computes predictive mean and (co)variance at Xnew, including mean_function
+        Does not make use of caching
+        """
+        mean, cov = self._conditional_fused(
+            Xnew, full_cov=full_cov, full_output_cov=full_output_cov
+        )
+        return self._add_mean_function(Xnew, mean), cov
+
+    @abstractmethod
+    def _conditional_fused(
+        self, Xnew, full_cov: bool = False, full_output_cov: bool = False
+    ) -> MeanAndVariance:
+        """
+        Computes predictive mean and (co)variance at Xnew, *excluding* mean_function
+        Does not make use of caching
+        """
+
+    def predict_f(
+        self, Xnew, full_cov: bool = False, full_output_cov: bool = False
+    ) -> MeanAndVariance:
+        """
+        Computes predictive mean and (co)variance at Xnew, including mean_function.
+        Relies on precomputed alpha and Qinv (see _precompute method)
+        """
+        if self.alpha is None or self.Qinv is None:
+            raise ValueError(
+                "Cache has not been precomputed yet. Call update_cache first or use fused_predict_f"
+            )
+        mean, cov = self._conditional_with_precompute(
+            Xnew, full_cov=full_cov, full_output_cov=full_output_cov
+        )
+        return self._add_mean_function(Xnew, mean), cov
+
+    @abstractmethod
```

Comment:
If it's abstract, why does it have an implementation?

Reply:
An oversight. I have removed the decorator.